dask/array/core.py

```diff
@@ -1156,6 +1156,12 @@ def __new__(cls, dask, name, chunks, dtype=None, meta=None, shape=None):
             if result is not None:
                 self = result
 
+        if hasattr(self, "dask") and name in self.dask.layers:
+            self.dask.layers[name].info["type"] = type(self)
+            self.dask.layers[name].info["chunk_type"] = type(meta)
+            self.dask.layers[name].info["chunks"] = chunks
+            self.dask.layers[name].info["dtype"] = dtype
+
```

Comment:
instead of this
```suggestion
        try:
            layer = self.dask.layers[name]
        except (AttributeError, KeyError):
            # self is no longer an Array after applying the plugins, OR
            # self has been replaced by a collection where dask is not a HighLevelGraph, OR
            # name is not the top layer's name (this can happen after the layer is manipulated,
            # to avoid a collision)
            pass
        else:
            layer.info.update({"type": type(self), "chunk_type": type(meta), "chunks": chunks, "dtype": dtype})

```